lib/grizzly/unsolicited_server/dtls_transport.ex

```diff
@@ -73,4 +48,43 @@ defmodule Grizzly.UnsolicitedServer.DTLSTransport do
   defp user_lookup(:psk, _username, userstate) do
     {:ok, userstate}
   end
+
+  defp do_listen(port, opts, attempts) do
+    ifaddr = Keyword.fetch!(opts, :ifaddr)
+
+    protocol =
+      case tuple_size(ifaddr) do
+        4 -> :inet
+        8 -> :inet6
+      end
+
+    listen_opts = [
+      protocol,
+      {:mode, :binary},
+      {:active, false},
+      {:verify, :verify_none},
+      {:versions, [:"dtlsv1.2", :dtlsv1]},
+      {:protocol, :dtls},
+      {:ciphers, [{:psk, :aes_128_cbc, :sha}]},
```

Comment:
The Dialyzer issue is due to using the old tuple way and not the map. This looks like the conversion function: https://github.com/erlang/otp/blob/master/lib/ssl/src/ssl.erl#L5013-L5017

```suggestion
      {:ciphers, [%{key_exchange: :psk, cipher: :aes_128_cbc, mac: :sha, prf: :default_prf}]},
```

Reply:
Great find, thanks!